client/fingerprint_manager.go

```diff
@@ -282,20 +283,69 @@ func (fm *FingerprintManager) watchDriverFingerprint(fpChan <-chan *drivers.Fing
 		case <-fm.shutdownCh:
 			cancel()
 			return
-		case fp := <-fpChan:
-			di := &structs.DriverInfo{
-				Attributes:        fp.Attributes,
-				Detected:          fp.Health != drivers.HealthStateUndetected,
-				Healthy:           fp.Health == drivers.HealthStateHealthy,
-				HealthDescription: fp.HealthDescription,
-				UpdateTime:        time.Now(),
-			}
-			fm.nodeLock.Lock()
-			n := fm.updateNodeFromDriver(name, di)
-			if n != nil {
-				fm.node = n
+		case fp, ok := <-fpChan:
+			// if the channel is closed attempt to open a new one
+			if !ok {
+				newFpChan, newCancel, err := fm.retryDriverFingerprint(name)
+				if err != nil {
+					fm.logger.Warn("failed to fingerprint driver, retrying in 30s", "error", err)
+					fm.nodeLock.Lock()
+					n := fm.updateNodeFromDriver(name, &structs.DriverInfo{
+						Healthy:           false,
+						HealthDescription: "failed to fingerprint driver",
+						UpdateTime:        time.Now(),
+					})
+					if n != nil {
+						fm.node = n
+					}
+					fm.nodeLock.Unlock()
+					time.Sleep(30 * time.Second)
```

Comment:
Can we do an exponential backoff with a much smaller starting point; thinking 5s: https://github.com/hashicorp/nomad/blob/master/client/allocrunner/taskrunner/task_runner.go#L1743